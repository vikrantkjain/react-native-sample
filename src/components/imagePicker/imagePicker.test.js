import MediaPicker from './mediaPicker';
import {Pressable} from 'react-native';
import renderer from 'react-test-renderer';

jest.mock('react-native-image-crop-picker', () => 'ImagePicker');
jest.mock('react-native-config', () => 'RNConfig');
describe('Test for MediaPicker', () => {
  const tree = renderer.create(
    <MediaPicker>
      <Pressable
        onPress={() => {
          ref.current.singlePhoto();
        }}
        testID="TEST-01"></Pressable>
    </MediaPicker>,
  );

  it('Render successfully', () => {
    expect(tree).toBeDefined();
  });
  it('Render correctly', () => {
    const treeToJSON = tree.toJSON();
    expect(treeToJSON).toMatchSnapshot();
  });
});
