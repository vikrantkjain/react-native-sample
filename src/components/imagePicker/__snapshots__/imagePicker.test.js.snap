// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for MediaPicker Render correctly 1`] = `
<View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    testID="TEST-01"
  />
  <Modal
    animationType="fade"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    style={
      {
        "flex": 1,
        "overflow": "hidden",
        "position": "absolute",
        "zIndex": 1,
      }
    }
    transparent={true}
    visible={false}
  />
</View>
`;
